src/Hl7.Fhir.Serialization/JsonNavigatorStream.cs

```diff
@@ -55,14 +61,25 @@ public static JsonNavigatorStream FromPath(string path)
         /// <param name="stream">A stream that returns a serialized json resource.</param>
         /// <param name="disposeStream">Determines if the <see cref="Dispose()"/> method should also dispose the specified <paramref name="stream"/> instance.</param>
         public JsonNavigatorStream(Stream stream, bool disposeStream)
+            : this(stream, disposeStream, null) { }
+
+        /// <summary>Create a new <see cref="JsonNavigatorStream"/> instance for the specified json resource stream.</summary>
+        /// <param name="stream">A stream that returns a serialized json resource.</param>
+        /// <param name="disposeStream">Determines if the <see cref="Dispose()"/> method should also dispose the specified <paramref name="stream"/> instance.</param>
+        /// <param name="settings">Parser settings that control de-serialization behavior.</param>
+        public JsonNavigatorStream(Stream stream, bool disposeStream, FhirJsonParsingSettings settings)
         {
             _stream = stream ?? throw Error.ArgumentNull(nameof(stream));
             _reader = SerializationUtil.JsonReaderFromStream(stream);
             _disposeStream = disposeStream;
+            ParserSettings = settings ?? FhirJsonParsingSettings.CreateDefault();
```

Comment:
Shouldn't this be `ParserSettings = settings.Clone()`?

Reply:
Agreed, the API should always clone incoming settings.
And we should ensure that all the settings classes have similar behavior.

Reply:
I've changed this behaviour to 
`ParserSettings = settings?.Clone() ?? FhirJsonParsingSettings.CreateDefault();`